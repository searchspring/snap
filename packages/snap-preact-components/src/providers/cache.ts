export { CacheProvider } from '@emotion/react';
import createCache from '@emotion/cache';

export const cache = createCache({
	key: 'ss',
	prepend: true,
});
