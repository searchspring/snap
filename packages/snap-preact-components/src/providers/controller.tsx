import { jsx } from '@emotion/react';
import preact, { h, createContext, ComponentChildren, ComponentType } from 'preact';
import { useContext } from 'preact/hooks';
import type { AbstractController } from '@searchspring/snap-controller';

const ControllerContext = createContext<null | AbstractController>(null);

export const ControllerProvider = ({ children, controller }: { children: ComponentChildren; controller: AbstractController }) => {
	return <ControllerContext.Provider value={controller}>{children}</ControllerContext.Provider>;
};

export const useController = () => useContext(ControllerContext);

export const withController = (Component: ComponentType) => (props: any) => {
	return <Component controller={useController()} {...props} />;
};
