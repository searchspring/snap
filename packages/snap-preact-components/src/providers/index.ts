export * from './store';
export * from './theme';
