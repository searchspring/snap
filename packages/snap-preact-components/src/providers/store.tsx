import preact, { h, createContext, ComponentChildren, ComponentType } from 'preact';
import { useContext } from 'preact/hooks';
import type { AbstractStore } from '@searchspring/snap-store-mobx';

const StoreContext = createContext<null | AbstractStore>(null);

export const StoreProvider = ({ children, store }: { children: ComponentChildren; store: AbstractStore }) => {
	return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};

export const useStore = () => useContext(StoreContext);

export function withStore<C extends ComponentType>(Component: C): C {
	return ((props: any) => <StoreContext.Consumer>{(store) => <Component {...props} store={store} />}</StoreContext.Consumer>) as C;
}
