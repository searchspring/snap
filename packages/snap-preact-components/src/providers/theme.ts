import { ThemeProvider as EmotionTheme } from '@emotion/react';
export { css, useTheme, withTheme } from '@emotion/react';

interface ThemeProviderProps {
	theme: Partial<Theme> | ((outerTheme: Theme) => Theme);
	children?: any;
}
interface ThemeProvider {
	(props: ThemeProviderProps): React.ReactElement;
}
export const ThemeProvider: ThemeProvider = EmotionTheme;

export const defaultTheme: Theme = {
	colors: {
		primary: '#3A23AD',
		secondary: '#00cee1',
		hover: '#f8f6fd',
		text: {
			secondary: '#ffffff',
		},
		message: {
			error: '#cc1212',
			warning: '#ecaa15',
			info: '#4c3ce2',
		},
	},
	components: {},
};
export interface Theme {
	variables?: {
		[key: string]: unknown;
	};
	colors?: {
		primary: string;
		secondary: string;
		hover?: string;
		text?: {
			secondary: string;
		};
		message?: {
			error: string;
			warning: string;
			info: string;
		};
	};
	components?: any;
	namedComponents?: any;
}
