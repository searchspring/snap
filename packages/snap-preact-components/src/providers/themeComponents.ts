import { BadgeProps } from '../components/Atoms/Badge';
import { BreadcrumbsProps } from '../components/Atoms/Breadcrumbs';
import { ButtonProps } from '../components/Atoms/Button';
import { DropdownProps } from '../components/Atoms/Dropdown';
import { FlexProps } from '../components/Atoms/Flex';
import { FormattedNumberProps } from '../components/Atoms/FormattedNumber';
import { IconProps } from '../components/Atoms/Icon';
import { ImageProps } from '../components/Atoms/Image';
import { LoadingBarProps } from '../components/Atoms/Loading';
import { BannerProps, InlineBannerProps } from '../components/Atoms/Merchandising';
import { OverlayProps } from '../components/Atoms/Overlay';
import { PriceProps } from '../components/Atoms/Price';
import { SkeletonProps } from '../components/Atoms/Skeleton';
import { StringProps } from '../components/Atoms/String';
import { TermsProps } from '../components/Atoms/Terms';
import { ResultLayoutProps } from '../components/Layouts/ResultLayout';
import { CarouselProps } from '../components/Molecules/Carousel';
import { CheckboxProps } from '../components/Molecules/Checkbox';
import { ErrorHandlerProps } from '../components/Molecules/ErrorHandler';
import { FacetGridOptionsProps } from '../components/Molecules/FacetGridOptions';
import { FacetHierarchyOptionsProps } from '../components/Molecules/FacetHierarchyOptions';
import { FacetListOptionsProps } from '../components/Molecules/FacetListOptions';
import { FacetPaletteOptionsProps } from '../components/Molecules/FacetPaletteOptions';
import { FacetSliderProps } from '../components/Molecules/FacetSlider';
import { FilterProps } from '../components/Molecules/Filter';
import { PaginationProps } from '../components/Molecules/Pagination';
import { PerPageProps } from '../components/Molecules/PerPage';
import { RatingProps } from '../components/Molecules/Rating';
import { ResultProps } from '../components/Molecules/Result';
import { SearchInputProps } from '../components/Molecules/SearchInput';
import { SelectProps } from '../components/Molecules/Select';
import { SlideoutProps } from '../components/Molecules/Slideout';
import { SortByProps } from '../components/Molecules/SortBy';
import { BranchOverrideProps } from '../components/Organisms/BranchOverride';
import { FacetProps } from '../components/Organisms/Facet';
import { FacetsProps } from '../components/Organisms/Facets';
import { FilterSummaryProps } from '../components/Organisms/FilterSummary';
import { NoResultsProps } from '../components/Organisms/NoResults';
import { ResultsProps } from '../components/Organisms/Results';
import { SearchHeaderProps } from '../components/Atoms/SearchHeader';
import { SidebarProps } from '../components/Organisms/Sidebar';
import { ToolbarProps } from '../components/Organisms/Toolbar';
import { AutocompleteProps } from '../components/Templates/Autocomplete';
import { RecommendationProps } from '../components/Templates/Recommendation';
import { SearchProps } from '../components/Templates/Search';

type GenericComponentProps<ComponentProps> = Partial<ComponentProps> & { named?: { [named: string]: Partial<ComponentProps> } };

export type ThemeComponents = {
	/* ATOMS */
	badge?: GenericComponentProps<BadgeProps>;
	breadcrumbs?: GenericComponentProps<BreadcrumbsProps>;
	button?: GenericComponentProps<ButtonProps>;
	dropdown?: GenericComponentProps<DropdownProps>;
	flex?: GenericComponentProps<FlexProps>;
	formattedNumber?: GenericComponentProps<FormattedNumberProps>;
	icon?: GenericComponentProps<IconProps>;
	image?: GenericComponentProps<ImageProps>;
	loadingBar?: GenericComponentProps<LoadingBarProps>;
	banner?: GenericComponentProps<BannerProps>;
	inlineBanner?: GenericComponentProps<InlineBannerProps>;
	overlay?: GenericComponentProps<OverlayProps>;
	price?: GenericComponentProps<PriceProps>;
	skeleton?: GenericComponentProps<SkeletonProps>;
	string?: GenericComponentProps<StringProps>;
	terms?: GenericComponentProps<TermsProps>;
	/* MOLECULES */
	carousel?: GenericComponentProps<CarouselProps>;
	checkbox?: GenericComponentProps<CheckboxProps>;
	errorHandler?: GenericComponentProps<ErrorHandlerProps>;
	facetGridOptions?: GenericComponentProps<FacetGridOptionsProps>;
	facetHierarchyOptions?: GenericComponentProps<FacetHierarchyOptionsProps>;
	facetListOptions?: GenericComponentProps<FacetListOptionsProps>;
	facetPaletteOptions?: GenericComponentProps<FacetPaletteOptionsProps>;
	facetSlider?: GenericComponentProps<FacetSliderProps>;
	filter?: GenericComponentProps<FilterProps>;
	pagination?: GenericComponentProps<PaginationProps>;
	perPage?: GenericComponentProps<PerPageProps>;
	rating?: GenericComponentProps<RatingProps>;
	result?: GenericComponentProps<ResultProps>;
	searchInput?: GenericComponentProps<SearchInputProps>;
	select?: GenericComponentProps<SelectProps>;
	slideout?: GenericComponentProps<SlideoutProps>;
	sortBy?: GenericComponentProps<SortByProps>;
	/* ORGANISMS */
	branchOverride?: GenericComponentProps<BranchOverrideProps>;
	facet?: GenericComponentProps<FacetProps>;
	facets?: GenericComponentProps<FacetsProps>;
	filterSummary?: GenericComponentProps<FilterSummaryProps>;
	noResults?: GenericComponentProps<NoResultsProps>;
	results?: GenericComponentProps<ResultsProps>;
	searchHeader?: GenericComponentProps<SearchHeaderProps>;
	sidebar?: GenericComponentProps<SidebarProps>;
	toolbar?: GenericComponentProps<ToolbarProps>;
	/* LAYOUTS */
	resultLayout?: GenericComponentProps<ResultLayoutProps>;
	/* TEMPLATES */
	autocomplete?: GenericComponentProps<AutocompleteProps>;
	recommendation?: GenericComponentProps<RecommendationProps>;
	search?: GenericComponentProps<SearchProps>;
};
