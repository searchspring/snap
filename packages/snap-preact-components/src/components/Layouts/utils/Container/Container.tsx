/** @jsx jsx */
import { ComponentChildren, h } from 'preact';

import { jsx, css } from '@emotion/react';
import classnames from 'classnames';
import { observer } from 'mobx-react-lite';

import { Theme, useTheme, CacheProvider } from '../../../../providers';
import { ComponentProps, StylingCSS } from '../../../../types';
import { mergeProps } from '../../../../utilities';

const _CSS = {
	flex: ({
		flex,
		flexDirection,
		textAlign,
		flexBasis,
		flexGrow,
		flexShrink,
		flexWrap,
		flexFlow,
		gap,
		rowGap,
		columnGap,
		item,
		justifyContent,
		alignItems,
		alignContent,
		alignSelf,
		width,
		height,
	}: ContainerProps) =>
		css({
			display: item ? undefined : 'flex',
			width,
			height,
			flexDirection,
			textAlign: textAlign ? (textAlign as any) : undefined,
			flexBasis: flexBasis ? (flexBasis as any) : undefined,
			flexGrow,
			flexShrink,
			flexWrap,
			flexFlow,
			gap,
			rowGap,
			columnGap,
			justifyContent,
			alignItems,
			alignContent,
			alignSelf,
			flex,
		}),
};

export const Container = observer((properties: ContainerProps): JSX.Element => {
	const globalTheme: Theme = useTheme();
	const defaultProps: Partial<ContainerProps> = {};

	const props = mergeProps('container', globalTheme, defaultProps, properties);

	const {
		children,
		flex,
		flexDirection,
		textAlign,
		flexBasis,
		flexGrow,
		flexShrink,
		flexWrap,
		flexFlow,
		gap,
		href,
		rowGap,
		columnGap,
		item,
		justifyContent,
		alignItems,
		alignContent,
		alignSelf,
		width,
		height,
		disableStyles,
		className,
		style,
	} = props;

	const styling: { css?: StylingCSS } = {};

	if (!disableStyles) {
		styling.css = [
			_CSS.flex({
				flex,
				flexDirection,
				textAlign,
				flexBasis,
				flexGrow,
				flexShrink,
				flexWrap,
				flexFlow,
				gap,
				rowGap,
				columnGap,
				item,
				justifyContent,
				alignItems,
				alignContent,
				alignSelf,
				width,
				height,
			}),
			style,
		];
	} else if (style) {
		styling.css = [style];
	}

	return (
		<CacheProvider>
			{href ? (
				<a {...styling} href={href} className={classnames(item ? 'ss__container-item' : 'ss__container', className)}>
					{children}
				</a>
			) : (
				<div {...styling} className={classnames(item ? 'ss__container-item' : 'ss__container', className)}>
					{children}
				</div>
			)}
		</CacheProvider>
	);
});

export interface ContainerProps extends ComponentProps {
	children?: ComponentChildren;
	width?: string | number;
	height?: string | number;
	alignItems?:
		| 'normal'
		| 'stretch'
		| 'center'
		| 'start'
		| 'end'
		| 'flex-start'
		| 'flex-end'
		| 'self-start'
		| 'self-end'
		| 'baseline'
		| 'first baseline'
		| 'last baseline'
		| 'safe center'
		| 'unsafe center'
		| 'inherit'
		| 'initial'
		| 'revert'
		| 'revert-layer'
		| 'unset';
	alignContent?:
		| 'center'
		| 'start'
		| 'end'
		| 'flex-start'
		| 'flex-end'
		| 'normal'
		| 'baseline'
		| 'first baseline'
		| 'last baseline'
		| 'space-between'
		| 'space-around'
		| 'space-evenly'
		| 'stretch'
		| 'safe center'
		| 'unsafe center'
		| 'inherit'
		| 'initial'
		| 'revert'
		| 'revert-layer'
		| 'unset';
	flexWrap?: 'nowrap' | 'wrap' | 'wrap-reverse';
	flexDirection?: 'row' | 'row-reverse' | 'column' | 'column-reverse';
	flexGrow?: number | 'inherit' | 'initial' | 'revert' | 'revert-layer' | 'unset';
	flexShrink?: number | 'inherit' | 'initial' | 'revert' | 'revert-layer' | 'unset';
	alignSelf?:
		| 'auto'
		| 'normal'
		| 'center'
		| 'start'
		| 'end'
		| 'self-start'
		| 'self-end'
		| 'flex-start'
		| 'flex-end'
		| 'baseline'
		| 'first baseline'
		| 'last baseline'
		| 'stretch'
		| 'safe center'
		| 'unsafe center'
		| 'inherit'
		| 'initial'
		| 'revert'
		| 'revert-layer'
		| 'unset';
	flexFlow?:
		| 'row'
		| 'row-reverse'
		| 'column'
		| 'column-reverse'
		| 'nowrap'
		| 'wrap'
		| 'wrap-reverse'
		| 'row nowrap'
		| 'column wrap'
		| 'column-reverse wrap-reverse'
		| 'inherit'
		| 'initial'
		| 'revert'
		| 'revert-layer'
		| 'unset';
	justifyContent?:
		| 'center'
		| 'start'
		| 'end'
		| 'flex-start'
		| 'flex-end'
		| 'left'
		| 'right'
		| 'normal'
		| 'space-between'
		| 'space-around'
		| 'space-evenly'
		| 'stretch'
		| 'safe center'
		| 'unsafe center'
		| 'inherit'
		| 'initial'
		| 'revert'
		| 'revert-layer'
		| 'unset';
	flex?: string | number;
	flexBasis?: string | number;
	textAlign?: 'left' | 'right' | 'center' | 'justify' | 'initial' | 'inherit';
	gap?: string | number;
	rowGap?: string | number;
	columnGap?: string | number;
	item?: boolean;
	href?: string;
}
