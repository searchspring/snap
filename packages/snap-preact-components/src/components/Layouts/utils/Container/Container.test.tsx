import { h } from 'preact';

import { render } from '@testing-library/preact';

import { Container } from './Container';
import { Theme, ThemeProvider } from '../../../../providers';

describe('Container Component', () => {
	it('renders', () => {
		const rendered = render(<Container />);
		const element = rendered.container.querySelector('div.ss__container');
		expect(element).toBeInTheDocument();
	});

	it('renders as item', () => {
		const rendered = render(<Container item={true} />);
		const element = rendered.container.querySelector('div.ss__container-item');
		expect(element).toBeInTheDocument();
	});

	it('renders with children', () => {
		const text = 'findMe';
		const children = (
			<div className="parent">
				<span className="child">{text}</span>
			</div>
		);
		const rendered = render(<Container>{children}</Container>);
		const element = rendered.container.querySelector('.ss__container');
		const parent = rendered.container.querySelector('.ss__container .parent');
		const child = rendered.container.querySelector('.ss__container .parent .child');

		expect(element).toBeInTheDocument();
		expect(parent).toBeInTheDocument();
		expect(child).toBeInTheDocument();
		expect(child).toHaveTextContent(text);
	});

	it('renders as link', () => {
		const link = 'www.href.com';
		const rendered = render(<Container href={link} />);
		const element = rendered.container.querySelector('a.ss__container');
		expect(element).toBeInTheDocument();
		expect(element).toHaveAttribute('href', link);
	});

	it('renders as link item', () => {
		const link = 'www.href.com';
		const rendered = render(<Container href={link} item={true} />);
		const element = rendered.container.querySelector('a.ss__container-item');
		expect(element).toBeInTheDocument();
		expect(element).toHaveAttribute('href', link);
	});

	it('renders as link with children', () => {
		const text = 'findMe';
		const link = 'www.href.com';
		const children = (
			<div className="parent">
				<span className="child">{text}</span>
			</div>
		);
		const rendered = render(<Container href={link}>{children}</Container>);

		const element = rendered.container.querySelector('a.ss__container');
		const parent = rendered.container.querySelector('a.ss__container .parent');
		const child = rendered.container.querySelector('a.ss__container .parent .child');

		expect(element).toBeInTheDocument();
		expect(parent).toBeInTheDocument();
		expect(child).toBeInTheDocument();
		expect(child).toHaveTextContent(text);
	});

	it('renders with classname', () => {
		const className = 'classy';
		const rendered = render(<Container className={className} />);
		const elem = rendered.container.querySelector('.ss__container');

		expect(elem).toHaveClass(className);
	});

	it('disables styles', () => {
		const rendered = render(<Container disableStyles />);
		const elem = rendered.container.querySelector('.ss__container');

		expect(elem?.classList).toHaveLength(1);
	});

	describe('styling props', () => {
		it('can set a width', () => {
			const prop = '15px';
			const rendered = render(<Container width={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`width: ${prop}`);
		});

		it('can set a height', () => {
			const prop = '15px';
			const rendered = render(<Container height={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`height: ${prop}`);
		});

		it('can set a alignItems', () => {
			const prop = 'center';
			const rendered = render(<Container alignItems={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`alignItems: ${prop}`);
		});

		it('can set a alignContent', () => {
			const prop = 'start';
			const rendered = render(<Container alignContent={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`alignContent: ${prop}`);
		});

		it('can set a flexWrap', () => {
			const prop = 'wrap';
			const rendered = render(<Container flexWrap={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`flexWrap: ${prop}`);
		});

		it('can set a flexDirection', () => {
			const prop = 'row';
			const rendered = render(<Container flexDirection={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`flexDirection: ${prop}`);
		});

		it('can set a flexGrow', () => {
			const prop = 2;
			const rendered = render(<Container flexGrow={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`flexGrow: ${prop}`);
		});

		it('can set a flexShrink', () => {
			const prop = 2;
			const rendered = render(<Container flexShrink={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`flexShrink: ${prop}`);
		});

		it('can set a alignSelf', () => {
			const prop = 'center';
			const rendered = render(<Container alignSelf={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`alignSelf: ${prop}`);
		});

		it('can set a flexFlow', () => {
			const prop = 'row';
			const rendered = render(<Container flexFlow={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`flexFlow: ${prop}`);
		});

		it('can set a justifyContent', () => {
			const prop = 'center';
			const rendered = render(<Container justifyContent={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`justifyContent: ${prop}`);
		});

		it('can set a flex', () => {
			const prop = '1';
			const rendered = render(<Container flex={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`flex: ${prop}`);
		});

		it('can set a flexBasis', () => {
			const prop = '1';
			const rendered = render(<Container flexBasis={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`flexBasis: ${prop}`);
		});

		it('can set a textAlign', () => {
			const prop = 'center';
			const rendered = render(<Container textAlign={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`textAlign: ${prop}`);
		});

		it('can set a gap', () => {
			const prop = '20px';
			const rendered = render(<Container gap={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`gap: ${prop}`);
		});

		it('can set a rowGap', () => {
			const prop = '20px';
			const rendered = render(<Container rowGap={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`rowGap: ${prop}`);
		});

		it('can set a columnGap', () => {
			const prop = '20px';
			const rendered = render(<Container columnGap={prop} />);
			const element = rendered.container.querySelector('.ss__container');
			expect(element).toBeInTheDocument();
			expect(element).toHaveStyle(`columnGap: ${prop}`);
		});
	});

	describe('Container theming works', () => {
		it('is themeable with ThemeProvider', () => {
			const globalTheme = {
				components: {
					container: {
						disableStyles: true,
					},
				},
			};
			const rendered = render(
				<ThemeProvider theme={globalTheme}>
					<Container height="300px" width="300px" />
				</ThemeProvider>
			);
			const elem = rendered.container.querySelector('.ss__container');
			expect(elem).toBeInTheDocument();
			expect(elem?.classList.length).toBe(1);
		});

		it('is themeable with theme prop', () => {
			const propTheme = {
				components: {
					container: {
						disableStyles: true,
					},
				},
			} as Theme;
			const rendered = render(<Container height="300px" width="300px" theme={propTheme} />);
			const elem = rendered.container.querySelector('.ss__container');
			expect(elem).toBeInTheDocument();
			expect(elem?.classList.length).toBe(1);
		});

		it('is theme prop overrides ThemeProvider', () => {
			const globalTheme = {
				components: {
					container: {
						disableStyles: false,
					},
				},
			};
			const propTheme = {
				components: {
					container: {
						disableStyles: true,
					},
				},
			} as Theme;
			const rendered = render(
				<ThemeProvider theme={globalTheme}>
					<Container height="300px" width="300px" theme={propTheme} />
				</ThemeProvider>
			);

			const elem = rendered.container.querySelector('.ss__container');
			expect(elem).toBeInTheDocument();
			expect(elem?.classList.length).toBe(1);
		});
	});
});
