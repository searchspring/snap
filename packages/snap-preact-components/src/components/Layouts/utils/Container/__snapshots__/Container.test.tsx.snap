// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component Container theming works is theme prop overrides ThemeProvider 1`] = `
<DocumentFragment>
  <div
    class="ss__container"
  />
</DocumentFragment>
`;

exports[`Container Component Container theming works is themeable with ThemeProvider 1`] = `
<DocumentFragment>
  <div
    class="ss__container"
  />
</DocumentFragment>
`;

exports[`Container Component Container theming works is themeable with theme prop 1`] = `
<DocumentFragment>
  <div
    class="ss__container"
  />
</DocumentFragment>
`;

exports[`Container Component disables styles 1`] = `
<DocumentFragment>
  <div
    class="ss__container"
  />
</DocumentFragment>
`;

exports[`Container Component renders 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component renders as item 1`] = `
<DocumentFragment>
  <div
    class="ss__container-item emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component renders as link 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<a
    class="ss__container emotion-0"
    href="www.href.com"
  />
</DocumentFragment>
`;

exports[`Container Component renders as link item 1`] = `
<DocumentFragment>
  <a
    class="ss__container-item emotion-0"
    href="www.href.com"
  />
</DocumentFragment>
`;

exports[`Container Component renders as link with children 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<a
    class="ss__container emotion-0"
    href="www.href.com"
  >
    <div
      class="parent"
    >
      <span
        class="child"
      >
        findMe
      </span>
    </div>
  </a>
</DocumentFragment>
`;

exports[`Container Component renders with children 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
    class="ss__container emotion-0"
  >
    <div
      class="parent"
    >
      <span
        class="child"
      >
        findMe
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Container Component renders with classname 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
    class="ss__container classy emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a alignContent 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: start;
  -ms-flex-line-pack: start;
  align-content: start;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a alignItems 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a alignSelf 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a columnGap 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-column-gap: 20px;
  column-gap: 20px;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a flex 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a flexBasis 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 1;
  -ms-flex-preferred-size: 1;
  flex-basis: 1;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a flexDirection 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a flexFlow 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row;
  -webkit-flex-flow: row;
  -ms-flex-flow: row;
  flex-flow: row;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a flexGrow 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a flexWrap 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a gap 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 20px;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a height 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 15px;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a justifyContent 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a rowGap 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  row-gap: 20px;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a textAlign 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: center;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;

exports[`Container Component styling props can set a width 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 15px;
}

<div
    class="ss__container emotion-0"
  />
</DocumentFragment>
`;
